components/Graficos/BarrasProducao.jsx

```diff
@@ -53,7 +53,7 @@ const GraficoBarrasProducao = ({
       { loading
         ? <Spinner theme='ColorSM' height='70vh' />
         : <ReactEcharts
-          option={ gerarOptions() }
+          option={ possuiDados? gerarOptions() : gerarGraficoSemDados() }
```

Comment:
@jvns2010 como você comentou, a mensagem de dados zerados apareceu em todas as instâncias do `GraficoBarrasProducao` na página de produção.

A sugestão que dou pra tentar resolver esse problema é setar a propriedade graphic com um objeto vazio `graphic: {}` nas opções geradas pela função `gerarOptions`, que é executada quando os dados não são gerados, e observar como as instâncias do gráfico se comportam. Me avisa sobre o resultado pra gente analisar melhor se precisar 😄

Reply:
A sua sugestão acabou não dando certo. No entanto, consegui encontrar o problema depois de fazer uma leitura mais extensa da [documentação](https://echarts.apache.org/en/api.html#echartsInstance.setOption). O problema aparece porque uma das opções padrão da biblioteca é a propriedade notMerge: false. Essa propriedade força o merge da nova option com a option antiga, nesse caso, como a primeira instância do gráfico era a instância sem dados, a biblioteca mergeava a instância inicial(sem dados) com a nova(com dados), gerando essa sobreposição. Ao inserir a prop notMerge: true, foi possível obter o comportamento esperado.

Reply:
Show!!